### ISSUE TITLE
How to input adj and features as pytorch Parameter to DGL?

### ISSUE THREAD

**Issue by monk1337** [dmlc/dgl#3367]
Hello, My GCN layer looks like this :

```
import torch
import numpy as np
import math

class GraphConvolution(torch.nn.Module):
    """
    Simple GCN layer, similar to https://arxiv.org/abs/1609.02907
    """

    def __init__(self, in_features, out_features, bias=False):
        super(GraphConvolution, self).__init__()
        self.in_features = in_features
        self.out_features = out_features
        self.weight = torch.nn.Parameter(torch.Tensor(in_features, out_features))
        if bias:
            self.bias = torch.nn.Parameter(torch.Tensor(1, 1, out_features))
        else:
            self.register_parameter('bias', None)
        self.reset_parameters()

    def reset_parameters(self):
        stdv = 1. / math.sqrt(self.weight.size(1))
        self.weight.data.uniform_(-stdv, stdv)
        if self.bias is not None:
            self.bias.data.uniform_(-stdv, stdv)

    def forward(self, input, adj):
        support = torch.matmul(input, self.weight)
        output  = torch.matmul(adj, support)
        if self.bias is not None:
            return output + self.bias
        else:
            return output

    def __repr__(self):
        return self.__class__.__name__ + ' (' \
               + str(self.in_features) + ' -> ' \
               + str(self.out_features) + ')'
```

For example if the Sample data is
```
feature_matrix = np.random.uniform(-1,1,[5,10])
adj                    = np.random.randint(0,2,[5,5])
```

The input to GCN layer are wrapped in Pytorch Parameter with `requires_grad = True` such as 

```
# where feature matrix is    [ nodes x features] numpy array
Graph_features  = torch.nn.Parameter(torch.from_numpy(feature_matrix).float(),requires_grad=True)

# where adj_d is adj file of [ nodes x nodes ] numpy array
A    = torch.nn.Parameter(torch.from_numpy(adj).float(),requires_grad=True)
```

```
gc1  = GraphConvolution(in_features = 10,  out_features = 10)
x = gc1(Graph_features, A)
```

While in DGL networks the input are in dgl.graph format How I can pass the input as Parameter like this to DGL network with `requires_grad=True`?

```
# where feature matrix is    [ nodes x features] numpy array
Graph_features  = torch.nn.Parameter(torch.from_numpy(feature_matrix).float(),requires_grad=True)

# where adj_d is adj file of [ nodes x nodes ] numpy array
A    = torch.nn.Parameter(torch.from_numpy(adj).float(),requires_grad=True)
```

**Comment by BarclayII** [comment]
Looks like you want to treat the adjacency matrix as a dense matrix?  In that case, you will need to pass in a complete graph as `dgl.graph` and simultaneously feed in the edge weights: 0 if there is no edge in the original graph, and 1 otherwise.  You then set the edge weight's `require_grad` to True.  DGL's `GraphConv` layer supports customized edge weights and you can compute the normalizer with `EdgeWeightNorm`.
